src/controllers/entry.controller.ts

```diff
@@ -178,37 +189,53 @@ export class GoodEntryController {
     const pogProfileParams: PogProfileParams = {
       userId: entry?.userId ?? '',
       email: entry?.email ?? '',
-      doGooder: entry?.doGooder ?? '',
+      doGooder: entry?.doGooder?.toLowerCase() ?? '',
     };
 
-    const pogProfile = await this.pogProfileService.findOrCreatePogProfile(
+    const pogProfile = await this.pogProfileRepository.findOrCreatePogProfile(
       pogProfileParams,
     );
 
-    const goodEntry = new GoodEntry(entry);
-    Object.assign(goodEntry, {
-      userId: pogProfile?.userId,
-      doGooder: pogProfile?.doGooder,
-    });
+    console.log('pogProfile:', pogProfile);
 
-    const entryData = await this.proofOfGoodSmartContractService.updateLedger(
-      'post',
-      goodEntry,
-    );
+    const goodEntry = new GoodEntry(entry);
+    Object.assign(goodEntry, pogProfile);
+    if (entry.externalId) {
+      const encodedExternalId = utils.formatBytes32String(entry.externalId);
+      Object.assign(goodEntry, {
+        externalId: encodedExternalId,
+      });
+    }
+    const persistGoodEntryData = new GoodEntry({
 
-    const response = await this.goodEntryRepository.create({
-      id: entryData.id,
-      doGooder: entryData.doGooder,
-      userId: pogProfile?.userId,
-      goodActivityId: entryData.goodActivityId,
-      goodOracleId: entryData.goodOracleId,
-      proofURL: entryData.proofURL,
-      value: entryData.value,
-      units: entryData.units,
-      timestamp: entryData.timestamp,
-      externalId: entryData.externalId,
+      doGooder: goodEntry?.doGooder?.toLowerCase(),
+      userId: goodEntry.userId,
+      email: entry.email,
+      goodActivityId: goodEntry.goodActivityId,
+      goodOracleId: goodEntry.goodOracleId,
+      goodPoints: goodEntry.goodPoints,
+      units: goodEntry.units,
+      timestamp: goodEntry.timestamp,
+      externalId: entry?.externalId,
     });
```

Comment:
could spread syntax be leveraged here?

Reply:
yeah I think normally I would, but since there's so many things going on in this code, thought it would be easier for whoever's reading it to clearly see which values I'm grabbing from where instead of having to back track to see what objects are holding which values